python/ml_croissant/ml_croissant/_src/structure_graph/nodes/source_test.py

```diff
@@ -65,3 +69,26 @@ def test_source_parses_list(json_ld, expected_source):
     assert Source.from_json_ld(issues, json_ld) == expected_source
     assert not issues.errors
     assert not issues.warnings
+
+
+@pytest.mark.parametrize(
+    ["value", "source", "expected_value"],
+    [
+        # No source
+        ["this is a value", None, "this is a value"],
+        # Capturing group
+        [
+            "train1234",
+            Source(apply_transform=(Transform(regex="(train|val)\\d\\d\\d\\d"),)),
+            "train",
+        ],
+        # Non working capturing group
+        [
+            "foo1234",
+            Source(apply_transform=(Transform(regex="(train|val)\\d\\d\\d\\d"),)),
+            "foo1234",
+        ],
```

Comment:
NIT: Perhaps it might be good to test for a chain of transforms as well?

Reply:
Acknowledged.

For the moment, we do not use this feature.